hack/build/ci/download-go-build-deps.sh

```diff
@@ -6,5 +6,5 @@ if ! [ -d ./third_party_licenses ]; then
 fi
 
 # fetch dependencies
-go get -d ./...
+go get -d ./...  # <-- WHY???
```

Comment:
on purpose? :)

Reply:
`./...` is used to get everything in sub directories as well, see [go docs](https://pkg.go.dev/cmd/go#hdr-Relative_import_paths), so it should be correct.

I don't think we need this script anymore at all, since we do `go mod download` as part of the `Dockerfile` -> `download-go-build-deps.sh` can be removed

Reply:
only the `go get` part as the licenses are still needed